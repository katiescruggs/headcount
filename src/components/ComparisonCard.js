import React from 'react';
import '../styles/ComparisonCard.css';

const ComparisonCard = (
  { districtOne, 
    districtTwo, 
    compareDistrictAverages}) => {

  const comparison = compareDistrictAverages(districtOne, districtTwo);

  return (
    <div className="comparison-card">
      <div className="comparison-card-head">
        <h2>AVERAGES</h2> 
      </div> 
      <p><strong>{districtOne} AVERAGE:</strong></p>
      <p><span className=
        "big-number district1-avg"> {comparison[districtOne]} </span></p>
      <p><strong>{districtTwo} AVERAGE: </strong></p>
      <p> <span className=
        "big-number district2-avg"> {comparison[districtTwo]} </span></p>
      <p><strong>COMPARISON RATIO:</strong></p>
      <p><span className=
        "big-number comparison"> {comparison.compared} </span> </p>
    </div>
  );
};

export default ComparisonCard; 