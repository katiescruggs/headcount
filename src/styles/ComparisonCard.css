.comparison-card {
    box-shadow: 1.5px 3px 3px 1.5px #566782;
    background: #ffffff;
    margin: 10px;
    width: 230px;
    height: 330px;
    text-align: center;
 
}

.comparison-card-head {
    display: flex;
    justify-content: center;
    background: #26517e;
    color: white;
    font-size: 15px;
    text-align: center;
    height: 60px;
}


.comparison-card-head h2 {
  font-size: 20px;
  margin-top: 10px;
}

.big-number {
  font-size: 30px;
  color: #7d7a7a
}