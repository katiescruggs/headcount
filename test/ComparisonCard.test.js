import React from 'react';
import { shallow, mount } from 'enzyme';

import ComparisonCard from '../src/components/ComparisonCard.js';

describe('ComparisonCard', () => {
  it('should have one h2 and six p tags', () => {
    const mockCompareAverages = (districtOne, districtTwo) => {
      return {[districtOne]: 'district one average', 
              [districtTwo]: 'district two average',
              compared: 'comparison ratio'  
            };
    };
    const wrapper = shallow(<ComparisonCard districtOne='COLORADO' districtTwo='ACADEMY 20' compareDistrictAverages={mockCompareAverages}/>)
    
    expect(wrapper.find('h2').length).toEqual(1);
    expect(wrapper.find('p').length).toEqual(6);
  });

  it('should display districtOne, districtTwo, and the comparison between them', () => {
    const mockCompareAverages = (districtOne, districtTwo) => {
      return {[districtOne]: 'district one average', 
              [districtTwo]: 'district two average',
              compared: 'comparison ratio'  
            };
    };
    const wrapper = shallow(<ComparisonCard districtOne='COLORADO' districtTwo='ACADEMY 20' compareDistrictAverages={mockCompareAverages}/>)
    expect(wrapper.find('.district1-avg').text()).toEqual('district one average');
    expect(wrapper.find('.district2-avg').text()).toEqual('district two average');
    expect(wrapper.find('.comparison').text()).toEqual('comparison ratio');
  });
});