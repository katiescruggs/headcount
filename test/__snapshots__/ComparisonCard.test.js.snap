exports[`ComparisonCard should match the snapshot 1`] = `
<div
  className="comparison-card">
  <div
    className="comparison-card-head">
    <h2>
      AVERAGES
    </h2>
  </div>
  <p>
    <strong>
      COLORADO
       AVERAGE:
    </strong>
  </p>
  <p
    className="big-number district1-avg">
    district one average
  </p>
  <p>
    <strong>
      ACADEMY 20
       AVERAGE:
    </strong>
  </p>
  <p
    className="big-number district2-avg">
    district two average
  </p>
  <p>
    <strong>
      COMPARISON RATIO:
    </strong>
  </p>
  <p
    className="big-number comparison">
    comparison ratio
  </p>
</div>
`;
